superset/migrations/versions/978245563a02_migrate_iframe_to_dash_markdown.py

```diff
@@ -0,0 +1,170 @@
+        iframes = session.query(Slice).filter_by(viz_type="iframe").all()
+        iframe_ids = [slc.id for slc in iframes]
+
+        for iframe in iframes:
+            iframe_params = json.loads(iframe.params or "{}")
+            url = iframe_params.get("url")
+            iframe_urls[iframe.id] = url
+
+        # find iframe viz_type that used in dashboard
+        dash_slc = (
+            session.query(dashboard_slices)
+            .filter(dashboard_slices.c.slice_id.in_(iframe_ids))
+            .all()
+        )
+        for entry in dash_slc:
+            dash_to_migrate[entry.dashboard_id].append(entry.slice_id)
+        dashboard_ids = list(dash_to_migrate.keys())
+
+        # replace iframe in dashboard metadata
+        dashboards = (
+            session.query(Dashboard).filter(Dashboard.id.in_(dashboard_ids)).all()
+        )
+        for i, dashboard in enumerate(dashboards):
+            print(
+                f"scanning dashboard ({i + 1}/{len(dashboards)}) dashboard: {dashboard.id} >>>>"
+            )
+
+            # remove iframe slices from dashboard
+            dashboard.slices = [
+                slc for slc in dashboard.slices if slc.id not in iframe_ids
+            ]
+
+            # find iframe chart position in metadata
+            # and replace it with markdown component
+            position_dict = json.loads(dashboard.position_json or "{}")
+            for key, value in position_dict.items():
+                if (
+                    value
+                    and isinstance(value, dict)
+                    and value["type"] == "CHART"
+                    and value["meta"]
+                    and value["meta"]["chartId"] in iframe_ids
+                ):
+                    iframe_id = value["meta"]["chartId"]
+                    # make new markdown component
+                    markdown = create_new_markdown_component(
+                        value, iframe_urls[iframe_id]
+                    )
+                    position_dict.pop(key)
+                    position_dict[markdown["id"]] = markdown
+
+                    # add markdown to layout tree
+                    parent_id = markdown["parents"][-1]
+                    children = position_dict[parent_id]["children"]
+                    children.remove(key)
+                    children.append(markdown["id"])
+
+                    dashboard.position_json = json.dumps(
+                        position_dict,
+                        indent=None,
+                        separators=(",", ":"),
+                        sort_keys=True,
+                    )
+                    session.merge(dashboard)
+
+        # remove iframe, separator and markup charts
+        session.query(Slice).filter(
+            Slice.viz_type.in_(["iframe", "separator", "markup"])
+        ).delete(synchronize_session=False)
+
+    except Exception as ex:
+        logging.exception(f"dashboard {dashboard.id} has error: {ex}")
```

Comment:
What state would we end in if only part of the iframes got deleted? I assume the charts would break because we're removing the backend api. Maybe it's not an issue though if the chance of an exception here is quite small